Dockerfile

```diff
@@ -33,7 +33,8 @@ RUN set -x \
  && /fossology/utils/fo-installdeps -e -y \
  && $_cleanup
 RUN curl -sS https://getcomposer.org/installer | php && \
-    mv composer.phar /usr/local/bin/composer
+	mkdir /tmp/composer && \
+    mv composer.phar /tmp/composer/composer
```

Comment:
But:
- we still do not know where it should be hosted
- the command `make composer_download` fails after composer was called once as root
- you need to download it once for every source tree
  - if you clone in a new location you might not be able to install the necessary composer, unless you remove the `~/.composer` folder. In fact you might not have the necessary permissions to do that and thus are not able build the project
- being forced to remove the folder `~/.composer` is not good
- you then have to adopt
  - the `fo-installdepts` script
  - the documentation: an additional step is needed for the source installation ([the archived documentation](http://archive15.fossology.org/projects/fossology/wiki/SysConfigSrc) which is still used by some people and is easily found will stop working)
  - the docker scripts
  - the vagrant scripts

> Using PATH is relying on environment to be correct in the first place.

- since you are the person defining `PATH=/my/composer/path:$PATH` that should not be a problem
- you might not want to prevent users from using system packages of composer. E.g. my operating system offers them.

> Also which variable can be used to declare that fossology tree ?

Do you search for an environmental variable used for this? I'm not aware of such an variable in the Makefiles

> But the solution is rather to remove that directory then in the Makefile at last step of the install phase (run by root as well)

You can't simply remove the `~/.composer` folder, which might be used for other projects on the machine.

> ...  as anyway it shouldn't be created as root.

That is an composer problem. When downloaded and installed as user, it seems to write the current home into the `composer.phar` which we rename to `composer`. Thus runnig this composer with root touches files in `~/.composer` of the user.